jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/model/instance/CaseFileInstanceImpl.java

```diff
@@ -218,11 +220,12 @@ public void setupRoles(Collection<CaseRole> roles) {
     }
     
     public void assignOwner(User actualOwner) {
-        boolean hasDefinedOwner = roles.containsKey(AuthorizationManager.OWNER_ROLE);
-        if (!hasDefinedOwner && !roles.isEmpty()) {
-            this.roles.put(AuthorizationManager.OWNER_ROLE, new CaseRoleInstanceImpl(AuthorizationManager.OWNER_ROLE, 1));
-            assign(AuthorizationManager.OWNER_ROLE, actualOwner);
+        CaseRoleInstance caseRoleInstance = this.roles.get(AuthorizationManager.OWNER_ROLE);
+        if (caseRoleInstance == null) {
+            caseRoleInstance = new CaseRoleInstanceImpl(AuthorizationManager.OWNER_ROLE, 1);
+            this.roles.put(AuthorizationManager.OWNER_ROLE, caseRoleInstance);
         }
+        ((CaseRoleInstanceImpl)caseRoleInstance).addRoleAssignment(actualOwner);
```

Comment:
this might throw exception of there are already owner assigned, is that what we want?